botocore/credentials.py

```diff
@@ -151,6 +164,11 @@ def __init__(self, access_key, secret_key, token,
         self._expiry_time = expiry_time
         self._time_fetcher = time_fetcher
         self.method = method
+        self.normalize()
+
+    def normalize(self):
+        self._access_key = botocore.compat.unicode(self._access_key, 'utf-8')
+        self._secret_key = botocore.compat.unicode(self._secret_key, 'utf-8')
```

Comment:
FYI. In case you may ask, why do we have to override the same name function in the base class, instead of reuse it. The reason is, super(...).normalize() will trigger the setter function of access_key and secret_key, which trigger _refresh() DURING initialization,  which will cause some existing test cases to fail.

Reimplementing it here is not too difficult. So I choose this route.